src/main/java/org/folio/circulation/domain/Item.java

```diff
@@ -145,17 +156,33 @@ public String getInstanceId() {
   }
 
   public String getCallNumber() {
-    return getProperty(holdingRepresentation, "callNumber");
+    return getEffectiveCallNumberValues(CALL_NUMBER_ID);
   }
 
   public String getCallNumberPrefix() {
-    return getProperty(holdingRepresentation, "callNumberPrefix");
+    return getEffectiveCallNumberValues(CALL_NUMBER_PREFIX_ID);
   }
 
   public String getCallNumberSuffix() {
-    return getProperty(holdingRepresentation, "callNumberSuffix");
+    return getEffectiveCallNumberValues(CALL_NUMBER_SUFFIX_ID);
+  }
+
+  private String getEffectiveCallNumberValues(String propertyName) {
```

Comment:
Thank you for improving the name. I have a minor follow up suggestion, as it only fetches a single property at a time, it could be singular, e.g. `getEffectiveCallNumberProperty` or `getEffectiveCallNumberValue`?

```suggestion
  private String getEffectiveCallNumberProperty(String propertyName) {
```

Reply:
fix